common/src/main/java/com/wynntils/core/features/overlays/Overlay.java

```diff
@@ -6,61 +6,71 @@
 
 import com.google.common.base.CaseFormat;
 import com.google.common.collect.ComparisonChain;
+import com.google.common.reflect.TypeToken;
 import com.mojang.blaze3d.platform.Window;
 import com.mojang.blaze3d.vertex.PoseStack;
 import com.wynntils.core.components.Managers;
 import com.wynntils.core.config.Config;
 import com.wynntils.core.config.ConfigHolder;
+import com.wynntils.core.config.RegisterConfig;
 import com.wynntils.core.features.AbstractConfigurable;
 import com.wynntils.core.features.Translatable;
 import com.wynntils.core.features.overlays.sizes.GuiScaledOverlaySize;
 import com.wynntils.core.features.overlays.sizes.OverlaySize;
+import com.wynntils.core.json.TypeOverride;
 import com.wynntils.utils.render.type.HorizontalAlignment;
 import com.wynntils.utils.render.type.VerticalAlignment;
+import java.lang.reflect.Type;
 import net.minecraft.client.renderer.MultiBufferSource;
 import net.minecraft.client.resources.language.I18n;
 import net.minecraft.world.phys.Vec2;
 
 public abstract class Overlay extends AbstractConfigurable implements Translatable, Comparable<Overlay> {
-    @Config(key = "overlay.wynntils.overlay.position", visible = false)
-    protected OverlayPosition position;
+    @RegisterConfig(key = "overlay.wynntils.overlay.position", visible = false)
+    public final Config<OverlayPosition> position = new Config<>(null);
```

Comment:
It is debatable if configs should be public or private, but let's not change it here.

Reply:
On the one hand it really doesn't matter since we do magical stuff and force our way past protections using reflection.

On the other hand, I think they should be public since we are in effect exposing them to the world using the config files.

Prior to this fix, it was a bit more scary to expose them since in theory anyone could have modified them. Now we can make them essentially read-only, which alleviates that problem.

But I changed them back to protected for now.